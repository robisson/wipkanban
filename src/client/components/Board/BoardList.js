// @flow
import * as React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import MiniBoard from "./MiniBoard";
import Tooltip from "@material-ui/core/Tooltip";
import AddIcon from "@material-ui/icons/AddCircle";
import IconButton from "@material-ui/core/IconButton";

type Props = {
  boards: Array<Object>
};

const BoardList = ({ boards }: Props) => {
  let boardsRendered = boards.map(board => (
    <MiniBoard key={board._id} {...board} />
  ));

  return (
    <div>
      <div>
        <Typography color="secondary" variant="h6" gutterBottom>
          <DeveloperBoardIcon /> My boards
          <Tooltip title="Add team">
            <IconButton color="default" aria-label="Add board">
              <AddIcon />
            </IconButton>
          </Tooltip>
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="wrap" spacing={0}>
        <Grid item lg={6}>
          {boardsRendered}
        </Grid>
      </Grid>
    </div>
  );
};

export default BoardList;
