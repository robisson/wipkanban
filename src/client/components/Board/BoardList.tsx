// @flow
import * as React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import MiniBoard from "./MiniBoard";
import Tooltip from "@material-ui/core/Tooltip";
import AddBoard from "./AddBoardContainer";

interface IBoardListProps {
  boards: Array<Object>;
}

const BoardList: React.StatelessComponent<any> = ({
  boards
}: IBoardListProps) => {
  let boardsRendered = boards.map((board: any) => (
    <Grid key={board._id} item lg={6}>
      <MiniBoard key={board._id} {...board} />
    </Grid>
  ));

  return (
    <div>
      <div>
        <Typography color="secondary" variant="h6" gutterBottom>
          <DeveloperBoardIcon /> My boards
          <Tooltip title="Add team">
            <AddBoard />
          </Tooltip>
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="wrap" spacing={8}>
        {boardsRendered}
      </Grid>
    </div>
  );
};

export default BoardList;
