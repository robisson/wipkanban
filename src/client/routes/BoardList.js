// @flow
import * as React from "react";
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import DeveloperBoardIcon from '@material-ui/icons/DeveloperBoard';
import MiniBoard from './MiniBoard';

const BoardList = () => {
  return (
    <div>
      <div>
        <Typography color="secondary" variant="title" gutterBottom>
          <DeveloperBoardIcon/>{' '}
          My boards
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="nowrap" spacing={0}>
        <Grid item lg={6}>
          <MiniBoard/>
        </Grid>
      </Grid>
    </div>
  )
};

export default BoardList;
