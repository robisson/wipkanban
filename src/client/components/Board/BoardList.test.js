import React from "react";
import BoardList from "./BoardList";
import { createShallow } from "@material-ui/core/test-utils";
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("BoardList Component", () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow();
  });

  it("should render a div", () => {
    let wrapper = shallow(<BoardList />);
    expect(wrapper.type()).toEqual("div");
  });

  it("should render 1 Typogragh and 2 Grid", () => {
    let wrapper = shallow(<BoardList />);
    expect(wrapper.find(Typography).length).toEqual(1);
    expect(wrapper.find(Grid).length).toEqual(2);
  });
});
