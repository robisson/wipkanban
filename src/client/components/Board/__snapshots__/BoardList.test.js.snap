// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardList Component should render 1 Typogragh and 2 Grid 1`] = `
<div>
  <div>
    <WithStyles(Typography)
      color="secondary"
      gutterBottom={true}
      variant="title"
    >
      <pure(DeveloperBoardIcon) />
       My boards
      <WithStyles(Tooltip)
        title="Add team"
      >
        <WithStyles(IconButton)
          aria-label="Add board"
          color="default"
        >
          <pure(AddCircleIcon) />
        </WithStyles(IconButton)>
      </WithStyles(Tooltip)>
    </WithStyles(Typography)>
  </div>
  <WithStyles(Grid)
    container={true}
    justify="flex-start"
    spacing={0}
    wrap="wrap"
  >
    <WithStyles(Grid)
      item={true}
      lg={6}
    >
      <WithStyles(MediaControlCard)
        _id={12121212}
        columns={
          Array [
            Object {
              "cards": Array [
                Object {
                  "id": 31,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 4,
              "title": "A fazer",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [],
              "hColumns": Array [
                Object {
                  "cards": Array [],
                  "hColumns": Array [],
                  "id": 2,
                  "title": "Chamados",
                  "vColumns": Array [
                    Object {
                      "cards": Array [],
                      "hColumns": Array [],
                      "id": 4,
                      "title": "Parado",
                      "vColumns": Array [],
                    },
                    Object {
                      "cards": Array [
                        Object {
                          "id": 35,
                          "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                        },
                      ],
                      "hColumns": Array [],
                      "id": 5,
                      "title": "Executando",
                      "vColumns": Array [],
                    },
                  ],
                },
                Object {
                  "cards": Array [],
                  "hColumns": Array [],
                  "id": 3,
                  "title": "Projetos",
                  "vColumns": Array [
                    Object {
                      "cards": Array [
                        Object {
                          "id": 36,
                          "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                        },
                      ],
                      "hColumns": Array [],
                      "id": 41,
                      "title": "Parado",
                      "vColumns": Array [],
                    },
                    Object {
                      "cards": Array [],
                      "hColumns": Array [],
                      "id": 51,
                      "title": "Executando",
                      "vColumns": Array [],
                    },
                  ],
                },
              ],
              "id": 1,
              "title": "Executando",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [
                Object {
                  "id": 32,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 9,
              "title": "Em teste",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [
                Object {
                  "id": 33,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
                Object {
                  "id": 34,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 10,
              "title": "Concluído",
              "vColumns": Array [],
            },
          ]
        }
        createdAt="today 12:00pm"
        createdBy="Robisson Oliveira"
        cycleTime="10"
        key="12121212"
        leadTime="10"
        name="Minhas atividades"
        throuput="10"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`BoardList Component should render a div 1`] = `
<div>
  <div>
    <WithStyles(Typography)
      color="secondary"
      gutterBottom={true}
      variant="title"
    >
      <pure(DeveloperBoardIcon) />
       My boards
      <WithStyles(Tooltip)
        title="Add team"
      >
        <WithStyles(IconButton)
          aria-label="Add board"
          color="default"
        >
          <pure(AddCircleIcon) />
        </WithStyles(IconButton)>
      </WithStyles(Tooltip)>
    </WithStyles(Typography)>
  </div>
  <WithStyles(Grid)
    container={true}
    justify="flex-start"
    spacing={0}
    wrap="wrap"
  >
    <WithStyles(Grid)
      item={true}
      lg={6}
    >
      <WithStyles(MediaControlCard)
        _id={12121212}
        columns={
          Array [
            Object {
              "cards": Array [
                Object {
                  "id": 31,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 4,
              "title": "A fazer",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [],
              "hColumns": Array [
                Object {
                  "cards": Array [],
                  "hColumns": Array [],
                  "id": 2,
                  "title": "Chamados",
                  "vColumns": Array [
                    Object {
                      "cards": Array [],
                      "hColumns": Array [],
                      "id": 4,
                      "title": "Parado",
                      "vColumns": Array [],
                    },
                    Object {
                      "cards": Array [
                        Object {
                          "id": 35,
                          "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                        },
                      ],
                      "hColumns": Array [],
                      "id": 5,
                      "title": "Executando",
                      "vColumns": Array [],
                    },
                  ],
                },
                Object {
                  "cards": Array [],
                  "hColumns": Array [],
                  "id": 3,
                  "title": "Projetos",
                  "vColumns": Array [
                    Object {
                      "cards": Array [
                        Object {
                          "id": 36,
                          "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                        },
                      ],
                      "hColumns": Array [],
                      "id": 41,
                      "title": "Parado",
                      "vColumns": Array [],
                    },
                    Object {
                      "cards": Array [],
                      "hColumns": Array [],
                      "id": 51,
                      "title": "Executando",
                      "vColumns": Array [],
                    },
                  ],
                },
              ],
              "id": 1,
              "title": "Executando",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [
                Object {
                  "id": 32,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 9,
              "title": "Em teste",
              "vColumns": Array [],
            },
            Object {
              "cards": Array [
                Object {
                  "id": 33,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
                Object {
                  "id": 34,
                  "title": "Criar layouts da tela principal do quadro. Este é um exemplo de tarefa ou card do quadro kanban",
                },
              ],
              "hColumns": Array [],
              "id": 10,
              "title": "Concluído",
              "vColumns": Array [],
            },
          ]
        }
        createdAt="today 12:00pm"
        createdBy="Robisson Oliveira"
        cycleTime="10"
        key="12121212"
        leadTime="10"
        name="Minhas atividades"
        throuput="10"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
